### PR TITLE
Add docs for opening ports on Kubernetes

### PR DESCRIPTION
**Pull request by hemildesai**
Add docs for https://github.com/skypilot-org/skypilot/pull/2713

### CODE REVIEW

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @hemildesai! I've reworked the docs to focus on how to get configured quickly with direct instructions - let me know what you think!

**Review by Michaelvll — COMMENTED** [review]
Thanks for adding the doc @hemildesai @romilbhardwaj! Left several minor comments.

File: docs/source/reference/config.rst [hunk 1]
```diff
@@ -173,6 +173,35 @@ Available fields and semantics:
...
+    # to be assigned so there might be a slight delay before the port can be accessed.
+    #
+    # ingress: Creates an ingress and a ClusterIP service for each port opened.
+    # Requires an ingress controller to be configured on the Kubernetes cluster.
+    # Currently, only the NGINX ingress controller is supported.
+    # Refer to https://kubernetes.github.io/ingress-nginx/deploy/ for details on
+    # deploying the NGINX ingress controller. In this mode, the port can be
+    # accessed externally using the Ingress URL plus a path prefix of the form
+    # /skypilot/{cluster_pod_name}/{port}
+    # Refer to kubernetes-ingress.yml.j2 for the exact template.
```
[the remark this answers is not in the document]
Comment (romilbhardwaj) [reply docs/source/reference/config.rst]: Thanks! I think this was a little too much detail, and the user should likely not look/edit at this j2 unless they really know what they are doing. I have removed this for now.

**Review by romilbhardwaj — COMMENTED** [review]
Thanks for the review @Michaelvll! PTAL : )

File: docs/source/reference/kubernetes/kubernetes-setup.rst [hunk 2]
```diff
@@ -215,6 +215,99 @@ Once the cluster is deployed and you have placed your kubeconfig at ``~/.kube/co
...
+
+
+By default, SkyPilot creates a `LoadBalancer Service <https://kubernetes.io/docs/concepts/services-networking/service/>`__ on your Kubernetes cluster to expose the port.
+
+If your cluster does not support LoadBalancer services, SkyPilot can also use `an existing Nginx IngressController <https://kubernetes.github.io/ingress-nginx/>`_ to create an `Ingress <https://kubernetes.io/docs/concepts/services-networking/ingress/>`_ to expose your service.
+
+.. _kubernetes-loadbalancer:
+
+LoadBalancer Service
+^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply docs/source/reference/kubernetes/kubernetes-setup.rst:244]: Oh, I see, let's keep the `^^^` then : )

File: docs/source/reference/kubernetes/kubernetes-setup.rst [hunk 3]
```diff
@@ -215,6 +215,99 @@ Once the cluster is deployed and you have placed your kubeconfig at ``~/.kube/co
...
+    The default LoadBalancer implementation in EKS selects a random port from the list of opened ports for the
+    `LoadBalancer's health check <https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html>`_. This can cause issues if the selected port does not have a service running behind it.
+
+
+    For example, if a SkyPilot task exposes 5 ports but only 2 of them have services running behind them, EKS may select a port that does not have a service running behind it and the LoadBalancer will not pass the healthcheck. As a result, the service will not be assigned an external IP address.
+
+    To work around this issue, make sure all your ports have services running behind them.
+
+.. note::
+    LoadBalancer services are not supported on kind clusters created using :code:`sky local up`.
```
Comment (Michaelvll) [inline docs/source/reference/kubernetes/kubernetes-setup.rst:271]: Just a quick thought, maybe we can do it in the future, but is it possible that we can have `sky local up` with the nginx ingress setup by default, so the user can directly try the open ports on the cluster that is `local up`'ed

**Review by Michaelvll — APPROVED** [review]
Thanks for the update! LGTM.

File: docs/source/reference/kubernetes/kubernetes-setup.rst (hunk 3 above)
Comment (romilbhardwaj) [reply docs/source/reference/kubernetes/kubernetes-setup.rst:271]: Good idea! tracking here - https://github.com/skypilot-org/skypilot/issues/3112 : )

**Pull request merged**